src/lib/logical_query_plan/functional_dependency.cpp

```diff
@@ -83,16 +92,56 @@ std::vector<FunctionalDependency> merge_fds(const std::vector<FunctionalDependen
 
       return true;
     });
-    if (existing_fd == fds_merged.end()) {
-      fds_merged.push_back(fd_to_add);
+    if (existing_fd == deflated_fds.end()) {
+      deflated_fds.push_back(fd_to_add);
     } else {
       // An FD with the same determinant expressions already exists. Therefore, we only have to add to the dependent
       // expressions set
       existing_fd->dependents.insert(fd_to_add.dependents.begin(), fd_to_add.dependents.end());
     }
   }
 
-  return fds_merged;
+  return deflated_fds;
+}
+
+std::vector<FunctionalDependency> merge_fds(const std::vector<FunctionalDependency>& fds_a,
```

Comment:
I don't like the verb "merge" now that we also have "intersect". I feel like "merge" could mean either a union or an intersection.

Can't come up with anything better though.

Reply:
Good point. What about `union_fds()` then?